File: versions/forge1201/src/common/java/com/idkidknow/mcreallink/forge1201/mixin/complement/BroadcastingMessage.java
package com.idkidknow.mcreallink.forge1201.mixin.complement;

import net.minecraft.network.chat.Component;

import java.util.function.Consumer;

public class BroadcastingMessage {
    private static final ThreadLocal<Boolean> ignoreMessage = ThreadLocal.withInitial(() -> false);
    public static boolean isIgnoreMessage() {
        return ignoreMessage.get();
    }
    private static Consumer<Component> callback = component -> {};
    public static Consumer<Component> getCallback() {
        return callback;
    }
    public static void setCallback(Consumer<Component> callback) {
        BroadcastingMessage.callback = callback;
    }

    public static void ignoreTemporarily(Runnable action) {
        ignoreMessage.set(true);
        action.run();
        ignoreMessage.set(false);
    }
}

File: mixin/src/main/java/com/idkidknow/mcrealcomm/mixin/complement/BroadcastingMessage.kt
package com.idkidknow.mcrealcomm.mixin.complement

import net.minecraft.network.chat.Component

/** @see com.idkidknow.mcrealcomm.mixin.mixin.PlayerListMixin */
object BroadcastingMessage {
    private val _ignoreMessage = ThreadLocal.withInitial { false }
    val ignoreMessage: Boolean get() = _ignoreMessage.get()
    var callback: (Component) -> Unit = {}

    fun ignoreTemporarily(action: () -> Unit) {
        _ignoreMessage.set(true)
        action()
        _ignoreMessage.set(false)
    }
}
